src/ert/run_models/base_run_model.py

```diff
@@ -579,11 +563,8 @@ def _build_ensemble(
     @property
     def paths(self) -> List[str]:
         run_paths = []
-        number_of_iterations = self.number_of_iterations
         active_realizations = np.where(self.active_realizations)[0]
-        for iteration in range(
-            self.start_iteration, self.start_iteration + number_of_iterations
-        ):
+        for iteration in range(self.start_iteration, self._total_iterations):
```

Comment:
Doesnt this change behavior?

Reply:
| Restart   | Start iter |  number_of_iterations  | phaseCount  
| --- | --- | --- | --- |
| -  | 0  | 4  | 4  |
| 0  | 1  | 3  | 4  |
| 1  | 2  | 2  | 4  |
| 2  | 3  | 1  | 4  |